File: Programmers/practice11.py
def solution(n):
    answer = n - 1
    prime = [True] * (n + 1)
    for i in range(2, int(n ** (1 / 2)) + 1):
        if prime[i] == True:
            for j in range(i * i, n + 1, i):
                if j % i == 0 and prime[j]:
                    prime[j] = False
                    answer -= 1

    return answer

n = 10
# answer = 4
print(solution(n))

n = 5
# answer = 3
print(solution(n))

File: Programmers/practice11.js
function solution(n) {
    const primes = Array(n + 1).fill(true);
    var answer = n - 1;
    for(let i = 2; i <= Math.sqrt(n); i++){
        if(primes[i]){
            for(let j = i * i; j <= n; j += i){
                if(j % i === 0 && primes[j]){
                    primes[j] = false;
                    answer--;
                }
            }
        }
    }

    return answer;
}